.github/workflows/release.yml

```diff
@@ -0,0 +1,74 @@
+name: Upload Release Assets
+
+on:
+  release:
+    types: [published]
+  workflow_dispatch:
+    inputs:
+      tag:
+        description: 'Tag to release'
+        required: true
+        type: string
+      action_id:
+        description: 'Action ID of the CI run'
+        required: true
+        type: string
+
+permissions:
+  contents: write
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: get latest ci id
```

Comment:
So, the problem I think I see with these steps is there is no check anywhere if the latest CI run is a tag, or if we're even running in a `tag` at all.

To me, it just looks like it's going to try automatically uploading binaries *every single CI run*. I don't know how safe that is.

Here's an example of our Docker publish step, we perform actions based on whether we're in a tag or not: https://github.com/girlbossceo/conduwuit/blob/main/.github/workflows/ci.yml#L511

In this case, I only want to run this if I'm running the workflow from a tag.

I also don't really see anything that replaces existing binaries uploaded. So, if I run a workflow, and re-run it, will it upload two binaries?

Can some safety precautions be added?

Reply:
> every single CI run

This workflow is triggered by publishing, it does not run on every commit.
```yaml
on:
  release:
    types: [published]
```
This means that it will only run if you are on the releases page, click publish release, and select a tag.

> So, if I run a workflow, and re-run it, will it upload two binaries?

No, if there is already a file with the same name in the release, it will be overwritten. It is done by the clobber of gh cli.
```
gh release upload $TAG "$file" --clobber --repo="${{github.repository}}"
```
[This document](https://cli.github.com/manual/gh_release_upload) describes the function of clobber.

> Can some safety precautions be added?

Of course, I can make the workflow exit with an error when it is triggered by commit. Or I can make it run only when the workflow is triggered by publishing, so that even if the workflow is accidentally triggered, there will be no problems.

Reply:
Okay then I likely misunderstood then. I guess I can just try it out as is and see what happens.

Reply:
You can just enter the action id, and v0.4.7-rc and run it manually, it will upload the action artifact you selected to v0.4.7-rc. You can also run it repeatedly to see if it replaces.